filterUsers.js

```diff
@@ -0,0 +1,124 @@
+let users = [
+    {
+        userName: "Mario Rossi",
+        competenceUser: [
+            {
+                nameComp: 'java',
+                area: 'linguaggi programmazione',
+                level: 3
+            },
+            {
+                nameComp: 'Linux',
+                area: 'Sistemi operativi',
+                level: 2
+            }
+        ]
+    },
+    {
+        userName: "Franco Verdi",
+        competenceUser: [
+            {
+                nameComp: 'js',
+                area: 'linguaggi programmazione',
+                level: 4
+            },
+            {
+                nameComp: 'Windows',
+                area: 'Sistemi operativi',
+                level: 1
+            }
+        ]
+    }
+];
+
+
+
+
+
+function nameORCompetence() {
+    console.log(document.getElementById("initialChoice").value)
+    if (document.getElementById("initialChoice").value == "name") {
```

Comment:
Hai considerato la possibiltà di usare uno switch? potrebbe rendere più leggero il carico visivo del codice, in molti casi gli if a cascata sono una pessima scelta architetturale.